File: src/packets/PacketC2SZoneName.js
const PacketC2S = require("./PacketC2S");

/**
 * Client -> Server
 * ID = 0x01
 * Zone Name
 * Renames a zone
 * Data:
 *      (Unsigned Char) Controller ID
 *      (Unsigned Char) Zone ID
 *      (NT String) New Zone Name
 */
class PacketC2SZoneName extends PacketC2S {
    parseData() {
        this._ctrllrID = this._buffer.readUInt8();
        this._zoneID = this._buffer.readUInt8();
        this._name = this._buffer.readStringNT();
    }

    getControllerID() {
        return this._ctrllrID;
    }

    getZoneID() {
        return this._zoneID;
    }

    getName() {
        return this._name;
    }

    getID() {
        return PacketC2SZoneName.ID;
    }
}

PacketC2SZoneName.ID = 0x01;

module.exports = PacketC2SZoneName;

File: src/RNetPi.Core/Packets/PacketC2SZoneName.cs
namespace RNetPi.Core.Packets;

/// <summary>
/// Client -> Server
/// ID = 0x04
/// Zone Name
/// Renames a zone
/// Data:
///     (Unsigned Char) Controller ID
///     (Unsigned Char) Zone ID
///     (NT String) New Zone Name
/// </summary>
public class PacketC2SZoneName : PacketC2S
{
    public const byte ID = 0x04;

    public byte ControllerID { get; private set; }
    public byte ZoneID { get; private set; }
    public string Name { get; private set; } = string.Empty;

    public PacketC2SZoneName(byte[] data) : base(data)
    {
    }

    public override byte GetID() => ID;

    protected override void ParseData()
    {
        ControllerID = Reader.ReadByte();
        ZoneID = Reader.ReadByte();
        Name = ReadNullTerminatedString();
    }
}